src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -82,96 +84,58 @@ class ProductController extends FrameworkBundleAdminController
-        /* @var $pagePreference AdminPagePreferenceInterface */
-        if ($pagePreference->getTemporaryShouldUseLegacyPage('product')) {
-            $legacyUrlGenerator = $this->get('prestashop.core.admin.url_generator_legacy');
-            /* @var $legacyUrlGenerator UrlGeneratorInterface */
-            $redirectionParams = array(
-                // do not transmit limit & offset: go to the first page when redirecting
-                'productOrderby' => $orderBy,
-                'productOrderway' => $sortOrder
-            );
-            return $this->redirect($legacyUrlGenerator->generate('admin_product_catalog', $redirectionParams));
-        }
-
-        // If POST, then check/cast POST params formats
-        if ($request->isMethod('POST')) {
-            foreach ($request->request->all() as $param => $value) {
-                switch ($param) {
-                    case 'filter_category':
-                        if (!is_numeric($value)) {
-                            $request->request->set($param, '');
-                        }
-                        if (is_numeric($value) && $value < 0) {
-                            $request->request->set($param, '');
-                        }
-                }
-            }
-        }
-
-        /* @var $logger LoggerInterface */
-        $logger = $this->get('logger');
+        $language = $this->getContext()->language;
+        $request->getSession()->set('_locale', $language->locale);
+        $request = $this->get('prestashop.adapter.filter_categories.request_purifier')->purify($request);
 
         /* @var $productProvider ProductInterfaceProvider */
         $productProvider = $this->get('prestashop.core.admin.data_provider.product_interface');
 
-        /* @var $translator TranslatorInterface */
-        $translator = $this->get('translator');
-
-        // get old values from persistence (before the current update)
+        // Set values from persistence and replace in the request
         $persistedFilterParameters = $productProvider->getPersistedFilterParameters();
+        $filterParametersUpdater = $this->get('prestashop.adapter.filter_parameters_updater');
 
-        if ($offset === 'last') {
-            $offset = $persistedFilterParameters['last_offset'];
-        }
-        if ($limit === 'last') {
-            $limit = $persistedFilterParameters['last_limit'];
-        }
-        if ($orderBy === 'last') {
-            $orderBy = $persistedFilterParameters['last_orderBy'];
-        }
-        if ($sortOrder === 'last') {
-            $sortOrder = $persistedFilterParameters['last_sortOrder'];
-        }
+        $filters = $filterParametersUpdater->setValues(
+            $persistedFilterParameters,
+            $offset,
+            $limit,
+            $orderBy,
+            $sortOrder
+        );
 
-        // override the old values with the new ones.
+        extract($filters, EXTR_OVERWRITE);
```

Comment:
Convenient method, but will it bring issues on IDE reporting variables as undefined?